From 9626654fa60825b5dc09a35bdfba6cb53d45e618 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@vmiklos.hu>
Date: Mon, 25 Jun 2018 19:49:54 +0200
Subject: [PATCH] configure: macOS improvement (#197)

* configure: fix shared library suffix on macOS

See <https://lists.gnu.org/archive/html/libtool/2010-04/msg00020.html>,
shrext_cmds is a command to be evaluated. Linux uses .so for both
loadable and shared libraries (in terms of libtool), so the problem is
not visible there, but on macOS this caused configure to literally look
for

libnspr4`test .$module = .yes && echo .so || echo .dylib`

for the nss backend, while nss provides libnspr4.dylib.
---
 configure.ac  |  5 ++++-
 src/xmldsig.c | 10 +++++-----
 src/xmlenc.c  |  4 ++--
 3 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2eb3f444..2adc625d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -175,7 +175,10 @@ dnl ==========================================================================
 dnl Hack for autoconf version mismatch
 dnl ==========================================================================
 if test "z$shrext" = "z" ; then
-    shrext=$shrext_cmds
+    AC_MSG_CHECKING(for shared library suffix)
+    module=no
+    eval shrext=$shrext_cmds
+    AC_MSG_RESULT($shrext)
 fi
 
 dnl ==========================================================================
-- 
2.16.4

